include/cantera/electron/WeakIonGasElectron.h

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file WeakIonGasElectron.h
+ * Header file for class WeakIonGasElectron.
+ */
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://www.cantera.org/license.txt for license and copyright information.
+
+#ifndef CT_WEAKIONGASELECTRON_H
+#define CT_WEAKIONGASELECTRON_H
+
+#include "cantera/electron/Electron.h"
+#include <Eigen/Sparse>
+
+typedef Eigen::SparseMatrix<double> SpMat;
+typedef Eigen::Triplet<double> T;
+
+namespace Cantera
+{
+/**
+ * This class calculates the properties of electron in a weakly ionized gas.
+ * Only electron-neutral collisions are considered for calculating the
+ * electron energy distribution function (EEDF).
+ *
+ * Reference:
+ * [1] G. J. M. Hagelaar and L. C. Pitchford
+ * "Solving the Boltzmann equation to obtain electron transport
+ * coefficients and rate coefficients for fluid models."
+ * Plasma Sources Science and Technology 14.4 (2005): 722.
+ * doi: https://doi.org/10.1088/0963-0252/14/4/011
+ * [2] A. Luque, "BOLOS: An open source solver for the Boltzmann equation,"
+ * https://github.com/aluque/bolos.
+ * [3] D. McElroy, C. Walsh, A. Markwick, M. Cordiner, K. Smith, T. Millar,
+ * "The umist database for astrochemistry 2012,"
+ * Astronomy & Astrophysics 550 (2013) A36.
+ * doi: https://doi.org/10.1051/0004-6361/201220465
+ * @ingroup electron
+ */
+class WeakIonGasElectron: public Electron
+{
+public:
+    WeakIonGasElectron();
+
+    virtual double electronDiffusivity();
+    virtual double electronMobility();
+    virtual double meanElectronEnergy();
+    virtual double powerGain();
+    virtual double elasticPowerLoss();
+    virtual double inelasticPowerLoss();
+    virtual double totalCollisionFreq();
+    virtual double inverseRateCoefficient(size_t k);
+    virtual double rateCoefficient(size_t k);
+    virtual void getNetPlasmaProductionRates(double* wdot);
+
+    //! The real part of the mobility. This is used in power gain for case of AC.
+    double realMobility();
+
+    //! electron temperature
+    //! If the reduced electric field is set, electron tempeature is calculated
+    //! from EEDF.
+    virtual double electronTemperature();
```

Comment:
Is it correct that `electronTemperature` is part of the state of the Plasma? If so, wouldn't it make sense to implement `Plasma` as a new ThermoPhase-derived object, instead of creating a completely separate interface?

Reply:
electronTemperature is the calculated parameter from the state EEDF (electron energy distribution function). I think that class PlasmaElectron is very different from thermoPhase.

Reply:
I understand that it is different from currently implemented `ThermoPhase` objects. I believe the question is whether it makes sense to not use the existing infrastructure, rather than tweaking the existing interfaces to accommodate your additions.